Mathlib/Analysis/Normed/Affine/AsymptoticCone.lean

```diff
@@ -0,0 +1,428 @@
+/-
+Copyright (c) 2025 Attila Gáspár. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Attila Gáspár
+-/
+
+import Mathlib.Analysis.Convex.Between
+import Mathlib.Analysis.Normed.Module.FiniteDimension
+
+/-!
+# Asymptotic cone of a set
+
+This file defines the asymptotic cone of a set in a normed affine space. The asymptotic cone of a
+set $A$ is usually defined as the set of points $v$ for which there exist sequences $t_n > 0$ and
+$x_n \in A$ such that $t_n \to 0$ and $t_n x_n \to v$. We take a different approach here using
+filters: we define the asymptotic cone of `s` as the set of vectors `v` such that
+`∃ᶠ p in Filter.atTop • 𝓝 v, p ∈ s` holds.
+
+## Main definitions
+
+* `asymptoticNhds`: the filter of neighborhoods at infinity in some direction.
+* `asymptoticCone`: the asymptotic cone of a subset of a normed affine space.
+
+## Main statements
+
+* `Convex.smul_vadd_mem_of_isClosed_of_mem_asymptoticCone`: if `v` is in the asymptotic cone of a
+closed convex set `s`, then every ray of direction `v` starting from `s` is contained in `s`.
+* `Convex.smul_vadd_mem_of_mem_nhds_of_mem_asymptoticCone`: if `v` is in the asymptotic cone of a
+convex set `s`, then every ray of direction `v` starting from the interior of `s` is contained in
+`s`.
+* `isBounded_iff_asymptoticCone_subset_singleton`: in finite dimensions, the asymptotic cone is
+trivial if and only if the set is bounded.
+-/
+
+open scoped Pointwise Topology
+open Filter
+
+section General
+
+variable
```

Comment:
Do you really need the `NormedSpace` and `MetricSpace` instances? Couldn't it work with continuity of `smul` only?

Reply:
Mathlib currently does not have the notion of a topological affine space, so we need to use `NormedAddTorsor`. But the proofs don't use norms, so that they can be easily generalized later.

Reply:
Isn't "topological add torsor" exactly `ContinuousVAdd`?

Reply:
Oh right, I see